extensions/hacker-news-500/src/view-500.tsx

```diff
@@ -0,0 +1,93 @@
+import { getFavicon } from "@raycast/utils";
+import { getStories } from "./hackernews";
+import { Story } from "./types";
+
+// Stories that the user clicked on
+const readCache = new Cache();
+const key = "read-stories";
+const read = JSON.parse(readCache.get(key) ?? "[]") as string[];
+// Stories that we've seen over the past week.
+// If we've seen something more than 24 hours ago, we remove it
+// This allows for stories posted e.g. 3 days ago to hit 500 points today
+// [{ story: {}, seen: 1234567890 }]
+const seenCache = new Cache();
+const seenKey = "seen-stories";
+const seenStories = JSON.parse(seenCache.get(seenKey) ?? "[]") as { story: Story; seen: number }[];
+
+seenStories.forEach((seenStory) => {
+  // just log how long ago we saw the story
+  const now = Date.now();
+  const storyDate = new Date(seenStory.seen).getTime();
+  const diff = now - storyDate;
+  const diffInHours = Math.floor(diff / (1000 * 60 * 60));
+  console.log(`Seen story ${seenStory.story.title} (${seenStory.story.external_url}) ${diffInHours} hours ago`);
+});
+
+export default function Command() {
+  const [loading, setLoading] = useState(true);
+  const [stories, setStories] = useState<Story[]>([]);
+  const hasUnread = stories.some((story) => !read.includes(story.external_url));
+  const unreadCount = stories.filter((story) => !read.includes(story.external_url)).length;
+
+  useEffect(() => {
+    setLoading(true);
+    getStories()
+      .then((stories) => {
+        const now = Date.now();
+
+        // Get list of unseen stories (they aren't in the seenCache)
+        const unseenStories = stories.filter((story) => {
+          return !seenStories.find((seenStory) => seenStory.story.external_url === story.external_url);
+        });
+
+        const allStoriesSeen = [...seenStories, ...unseenStories.map((story) => ({ story, seen: now }))];
+        seenCache.set(seenKey, JSON.stringify(allStoriesSeen));
+        const twentyFourHoursInMs = 240 * 60 * 60 * 1000;
+        return allStoriesSeen
+          .filter(({ seen }) => {
+            const storyDate = new Date(seen).getTime();
+            return now - storyDate < twentyFourHoursInMs;
+          })
+          .map(({ story }) => story);
+      })
+      .then((stories) => setStories(stories))
+      .finally(() => setLoading(false));
+  }, []);
+
+  return (
+    <MenuBarExtra
+      icon={{
+        source: hasUnread ? "icon.png" : "icon-64-dark.png",
+        tintColor: hasUnread ? null : { light: "#000000", dark: "#ffffff", adjustContrast: true },
+      }}
+      tooltip={`Hacker News 500+ Stories${unreadCount > 0 ? ` (${unreadCount})` : ""}`}
+      isLoading={loading}
+    >
+      {stories.length === 0 && <MenuBarExtra.Item title="No recent stories" />}
+      {stories?.map((story: Story) => (
+        <MenuBarExtra.Item
+          key={story.external_url}
+          icon={
+            read.includes(story.external_url)
+              ? getFavicon(story.external_url)
```

Comment:
This is using the `external_url` which seems to always be a Hacker News URL, should it not be `url` instead?

Reply:
I used this because you can get to the story url from the HN url, but not the other way around. Often when a story gets to 500 I think it's worth reading the comments.

Reply:
Oh this is for the icon, sorry. The reason I did that is because sometimes the icons don't load and it does this weird flickering thing. I wanted to show an "unread" vs "read" state (based on whether you clicked on it) and wasn't really sure how to do that well without something custom (which I woulnd't be able to pull off)

Reply:
I see, thanks for the explanation. I thought it looked a little strange with a list full of Hacker News favicons. Perhaps you could use the dot as per the "unread" state and just have a grey one for when it's "read"?

Reply:
Yeah I think that looks better.

![CleanShot 2025-04-09 at 12 37 47@2x](https://github.com/user-attachments/assets/7af6a91f-1a54-4d56-8dfc-6c6da7b7bbd5)
![CleanShot 2025-04-09 at 12 35 24@2x](https://github.com/user-attachments/assets/bb7de1aa-c315-493f-9b0f-47a06093e0d9)